datafusion/functions/Cargo.toml

```diff
@@ -52,9 +52,9 @@ encoding_expressions = ["base64", "hex"]
 # enable math functions
 math_expressions = []
 # enable regular expressions
-regex_expressions = ["regex"]
+regex_expressions = ["regex", "string_expressions"]
```

Comment:
I don't understand the need for this change (which seems to basically make `regex_expressions` and `string_expressions` the same feature flag)

I see you added `datafusion/functions/src/regex/regexp_common.rs`, but that seems only to be used by string functions. Perhaps it would be beter placed somewhere in functions/src/ 🤔

Reply:
@alamb thanks for reviewing 🙏
> Perhaps it would be beter placed somewhere in functions/src/

oh, I agree, will update it soon